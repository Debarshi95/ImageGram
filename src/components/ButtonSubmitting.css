.buttonsubmitting__root {
  width: 100%;
  padding: 10px;
  background: #fafafa;
  border: 1.8px solid #b8b8b8;
  border-radius: 4px;
  margin-top: 6px;
  display: flex;
  justify-content: center;
}

.buttonsubmitting__root > button {
  background: transparent;
  width: 100%;
  outline: none;
  border: 0;
}

.buttonsubmitting__spinner {
  width: 20px !important;
  height: 20px !important;
}
