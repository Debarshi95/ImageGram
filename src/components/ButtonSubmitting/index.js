import { CircularProgress } from "@material-ui/core";
import React from "react";
import "./index.css";

function ButtonSubmitting({ submitting, disabled, text, handler }) {
  return (
    <>
      {submitting && (
        <div className="buttonsubmitting__loader">
          <CircularProgress />
        </div>
      )}

      {!submitting && (
        <button
          className="buttonsubmitting__button"
          type="click"
          disabled={disabled}
          onClick={handler}
        >
          {text}
        </button>
      )}
    </>
  );
}

export default ButtonSubmitting;
