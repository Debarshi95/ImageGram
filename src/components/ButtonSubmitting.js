import { CircularProgress } from "@material-ui/core";
import React from "react";
import "./ButtonSubmitting.css";

function ButtonSubmitting({ submitting, disabled, text, handler }) {
  return (
    <>
      {submitting && (
        <div className="buttonsubmitting__loader">
          <CircularProgress />
        </div>
      )}

      {!submitting && (
        <button
          className="buttonsubmitting__button"
          type="click"
          disabled={disabled}
          onClick={handler}
        >
          {text}
        </button>
      )}
    </>
  );
}

export default ButtonSubmitting;
