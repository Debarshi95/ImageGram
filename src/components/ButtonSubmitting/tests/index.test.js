import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ButtonSubmitting from "../index";

describe("<ButtonSubmitting/> tests", () => {
  let submitting;
  let disabled;
  let text;
  let handlerSpy;
  beforeEach(() => {
    submitting = false;
    disabled = true;
    text = "Signing in";
    handlerSpy = jest.fn();
  });
  it("should render and match the snapshot", () => {
    const { baseElement } = render(
      <ButtonSubmitting
        text={text}
        handler={handlerSpy}
        submitting={submitting}
        disabled={disabled}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("should render button when 'submitting' is false", () => {
    const { baseElement } = render(
      <ButtonSubmitting
        submitting={submitting}
        text={text}
        disabled={disabled}
      />
    );
    const btn = baseElement.querySelector(".buttonsubmitting__button");
    expect(btn).toBeInTheDocument();
    expect(btn.disabled).toBeTruthy();
    expect(btn.textContent).toEqual(text);
  });

  it("should render circularbar when 'submitting' is true", async () => {
    submitting = true;
    const { findByRole } = render(
      <ButtonSubmitting submitting={submitting} handler={handlerSpy} />
    );
    const bar = await findByRole("progressbar");

    expect(bar).toBeInTheDocument();
  });

  it("should call handlerSpy on click", async () => {
    const { baseElement } = render(
      <ButtonSubmitting submitting={submitting} handler={handlerSpy} />
    );
    const btn = baseElement.querySelector(".buttonsubmitting__button");
    fireEvent.click(btn);
    expect(handlerSpy).toBeCalled();
  });
});
