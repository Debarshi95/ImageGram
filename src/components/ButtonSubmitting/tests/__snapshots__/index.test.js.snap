// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSubmitting/> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <button
      class="buttonsubmitting__button"
      disabled=""
      type="click"
    >
      Signing in
    </button>
  </div>
</body>
`;
